Tasks/KubernetesManifestV0/package.json

```diff
@@ -6,7 +6,8 @@
     "@types/node": "^10.17.0",
     "@types/q": "^1.5.0",
     "@types/uuid": "^8.3.0",
-    "azure-pipelines-task-lib": "^3.0.6-preview.0",
+    "azure-pipelines-task-lib": "^4.1.0",
```

Comment:
when I run a pipeline with ARM SC, I face the error:
##[error]Cannot read property 'retrieveSecret' of undefined

It might be caused after bumping this version.
I tried to lower it to 3.0.6-preview.0, but still has the same error.

Another assumption is that azure-arm-rest-v2 is not compatible with node 10, therefore we should use v1 version (probably).

Reply:
On the other hand, when I run it with Kubernetes service connection, I have this error:
error: failed to create secret Post "http://localhost:8080/api/v1/namespaces/default/secrets?fieldManager=kubectl-create&fieldValidation=Strict": dial tcp [::1]:8080: connect: connection refused


Log from the task:
==============================================================================
			Kubectl Client Version: v1.26.0
	Kubectl Server Version: Could not find kubectl server version
==============================================================================
/usr/local/bin/kubectl delete secret dr secret --namespace default
E1223 13:24:29.329568    1701 memcache.go:238] couldn't get current server API group list: Get "[http://localhost:8080/api?timeout=32s"](http://localhost:8080/api?timeout=32s%22): dial tcp [::1]:8080: connect: connection refused
E1223 13:24:29.330023    1701 memcache.go:238] couldn't get current server API group list: Get "[http://localhost:8080/api?timeout=32s"](http://localhost:8080/api?timeout=32s%22): dial tcp [::1]:8080: connect: connection refused
E1223 13:24:29.331474    1701 memcache.go:238] couldn't get current server API group list: Get "[http://localhost:8080/api?timeout=32s"](http://localhost:8080/api?timeout=32s%22): dial tcp [::1]:8080: connect: connection refused
E1223 13:24:29.332835    1701 memcache.go:238] couldn't get current server API group list: Get "[http://localhost:8080/api?timeout=32s"](http://localhost:8080/api?timeout=32s%22): dial tcp [::1]:8080: connect: connection refused
E1223 13:24:29.334187    1701 memcache.go:238] couldn't get current server API group list: Get "[http://localhost:8080/api?timeout=32s"](http://localhost:8080/api?timeout=32s%22): dial tcp [::1]:8080: connect: connection refused
The connection to the server localhost:8080 was refused - did you specify the right host or port?
/usr/local/bin/kubectl create secret docker-registry dr secret --docker-username *** --docker-password *** --docker-server *** --docker-email ServicePrincipal@AzureRM --namespace default
error: failed to create secret Post "[http://localhost:8080/api/v1/namespaces/default/secrets?fieldManager=kubectl-create&fieldValidation=Strict"](http://localhost:8080/api/v1/namespaces/default/secrets?fieldManager=kubectl-create&fieldValidation=Strict%22): dial tcp [::1]:8080: connect: connection refused
##[error]error: failed to create secret Post "[http://localhost:8080/api/v1/namespaces/default/secrets?fieldManager=kubectl-create&fieldValidation=Strict"](http://localhost:8080/api/v1/namespaces/default/secrets?fieldManager=kubectl-create&fieldValidation=Strict%22): dial tcp [::1]:8080: connect: connection refused
Finishing: KubernetesManifest


Maybe this is the answer to the problem:
Kubectl Server Version: Could not find kubectl server version



For both service connections (ARM, kube) I used "create-secret" action.